### ISSUE TITLE
Custom dataset

### ISSUE THREAD

**Issue by woan11103** [PotatoBigRoom/CoherentGS#1]
Only a video as input, how to get final 3DGS step by step?

**Comment by PotatoBigRoom** [comment]
**1.Data Preprocessing**
The process begins with data preprocessing on the raw video input by extracting frames and filtering for sharp image sequences.
**2.Pose Estimation (SfM)**
Next is the Pose Estimation (SfM) stage, utilizing models such as COLMAP or VGGT to estimate the camera extrinsics (R, T) for each frame and generate an initial sparse point cloud, which serves as the geometric baseline for scene construction.
**3.Initialization**
This is followed by the initialization phase, where these sparse points are converted into 3D Gaussians and initialized with attributes including position, rotation, and scaling.
**4.Optimization Loop**
Upon this foundation, the system proceeds to the core Optimization Loop; through differentiable rasterization, the Gaussians are projected onto 2D views to compute the loss between the rendered images and the ground truth. Backpropagation is then employed to iteratively update the Gaussian parameters and perform adaptive density control.
**5.Export**
Finally, once the model converges, all Gaussian attributes are exported into a `.ply` file, enabling real-time 3D rendering and navigation.

**Comment by woan11103** [comment]
I see. Thank you. 
But my question is how to process our own data (like your provided dataset) from just a RGB Video.

**Comment by woan11103** [comment]
<img width="1497" height="561" alt="Image" src="https://github.com/user-attachments/assets/4bfbc462-1053-4399-a233-1cdaa61241f1" />

This is the output of your run_dl3v.sh. Is there any bug in your current script?

**Comment by woan11103** [comment]
<img width="500" height="577" alt="Image" src="https://github.com/user-attachments/assets/339f5c29-4f3f-4fcc-8a5c-f7385b326a28" />

**Comment by PotatoBigRoom** [comment]
If you have the complete RGB video, you can first obtain some frame images from it, and refer to steps 2, 3, 4, and 5 above. You can also refer to https://github.com/graphdeco-inria/gaussian-splatting

**Comment by WenhaoSong-1512** [comment]
> <img alt="Image" width="1497" height="561" src="https://private-user-images.githubusercontent.com/249259990/526978331-4bfbc462-1053-4399-a233-1cdaa61241f1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY2NDQ0NDcsIm5iZiI6MTc2NjY0NDE0NywicGF0aCI6Ii8yNDkyNTk5OTAvNTI2OTc4MzMxLTRiZmJjNDYyLTEwNTMtNDM5OS1hMjMzLTFjZGFhNjEyNDFmMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyNVQwNjI5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDMwNWNkZDM0NTAzZGI1NjZkNzRkMTIwOTc3YjI0OTZiNWM4MDcxMjNlZmRiYjI1NmNiZjM2ZGQ0Y2JkOGNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LS5MZY7u2s_xGKms1Eq_9fCVxWZwXbBWd_BxVwbKhn0">
> This is the output of your run_dl3v.sh. Is there any bug in your current script?

I got the same result after training. Have you solved it?
